File: native/test/java/com/rarnu/yugioh/NativeAPI.java
package com.rarnu.yugioh;

public class NativeAPI {
    static {
        System.loadLibrary("yugiohapi");
    }

    public static native String cardSearch(
        String aname, 
        String ajapname, 
        String aenname,
        String arace,
        String aelement,
        String aatk,
        String adef,
        String alevel,
        String apendulum,
        String alink,
        String alinkarrow,
        String acardtype,
        String acardtype2,
        String aeffect,
        int apage);
}

File: android/app/src/main/java/com/rarnu/yugioh/NativeAPI.kt
package com.rarnu.yugioh

object NativeAPI {
    init {
        System.loadLibrary("yugiohapi")
    }

    external fun cardSearch(
            aname: String,
            ajapname: String,
            aenname: String,
            arace: String,
            aelement: String,
            aatk: String,
            adef: String,
            alevel: String,
            apendulum: String,
            alink: String,
            alinkarrow: String,
            acardtype: String,
            acardtype2: String,
            aeffect: String,
            apage: Int): String


}
